model/jpa/src/main/java/org/keycloak/models/jpa/UserAdapter.java

```diff
@@ -316,6 +316,9 @@ public String getEmail() {
 
     @Override
     public void setEmail(String email) {
+        if (email == null || email.trim().isEmpty()) {
```

Comment:
Aren't there any StringUtils which allow to write these three lines as simple oneliner like email = StringUtils.trimmToNull(email); ?

Reply:
There isn't. But they are coming in a different PR. We can refactor that later.